### PR TITLE
Refactor: moving cqlsh and nodetool calls to run from BaseNode objects

### PR DESCRIPTION
**Pull request by bentsi**
Currently all over the coded cqlsh and nodetool are run by raw command using remoter object.
To prevent errors and handle Authentication cqlsh and nodetool command are now run from one place.
- Added 1 disruptive 2 nondisruptive Nemesis pipeline and YAML
- fixes
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [ ] ~~All new and existing unit tests passed (`hydra unit-tests`)~~
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: jenkins-pipelines/longevity-1tb-7days-1Dis-2NonDis-Nemesises.jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!groovy
```
Comment (fruch) [inline jenkins-pipelines/longevity-1tb-7days-1Dis-2NonDis-Nemesises.jenkinsfile]: how's this pipe related to the refactoring ?

Comment (bentsi) [reply jenkins-pipelines/longevity-1tb-7days-1Dis-2NonDis-Nemesises.jenkinsfile]: The refactor is done for this pipeline :)

File: sdcm/cluster.py [hunk 2]
```diff
@@ -1823,10 +1820,56 @@ def restart_node_with_resharding(self, murmur3_partitioner_ignore_msb_bits=12):
...
+            -h	--host	Hostname or IP address.
+            -p	--port	Port number.
+            -pwf	--password-file	Password file path.
+            -pw	--password	Password.
+            -u	--username	Username.
+        :param ignore_status: don't throw exception if the command fails
+        :return: Remoter result object
+        """
+        cmd = self._gen_nodetool_cmd(sub_cmd, args, options)
+        result = self.remoter.run(cmd, ignore_status=ignore_status)
```
Comment (amoskong) [inline sdcm/cluster.py]: `verbose=True` is used in executed some nodetool command for outputing the result in log, but the new function doesn't support this.

Is it expected?

Comment (aleksbykov) [reply sdcm/cluster.py]: verbose is True by default. So any command without verbose=False, will be logged.

Comment (amoskong) [reply sdcm/cluster.py]: You are right.

**Comment by bentsi** [comment]
backported to 3.1 fixes only
- aab603f - Creation of test stats should be after the db nodes provisioned (5 minutes ago) <Bentsi Magidovich>
- 8aef53c - Replacing retry_till_success by retrying decorator (5 minutes ago) <Bentsi Magidovich>
- 473be5d - Adding nodetool status after all DB nodes setup (5 minutes ago) <Bentsi Magidovich>
- 122a92a - Fixing repair_streaming_exists according to the new API (5 minutes ago) <Bentsi Magidovich>

**Comment by amoskong** [comment]
Hi @bentsi @aleksbykov 

This PR added auth support for cqlsh query in run_fullscan() & get_non_system_ks_cf_list() by https://github.com/scylladb/scylla-cluster-tests/pull/1066/commits/f6c2b806e42f5e850d3c11dea5e46042e9f47064 .

I saw fullscan in 2019.1 longevity job (authorization-and-tls-ssl enabled) failed for cqlsh auth issue.
This commit https://github.com/scylladb/scylla-cluster-tests/pull/1066/commits/f6c2b806e42f5e850d3c11dea5e46042e9f47064 isn't backported 3.1 & 2019.1. 

```
Exception in thread Thread-684:
Traceback (most recent call last):
  File "/usr/lib64/python2.7/threading.py", line 812, in __bootstrap_inner
    self.run()
  File "/usr/lib64/python2.7/threading.py", line 765, in run 
    self.__target(*self.__args, **self.__kwargs)
  File "/sct/sdcm/utils.py", line 117, in inner
    res = func(*args, **kwargs)
  File "/sct/sdcm/cluster.py", line 3017, in run_in_thread
    result = self.run_fullscan(ks_cf, loader_node, db_node.ip_address)
  File "/sct/sdcm/cluster.py", line 2975, in run_fullscan
    ks_cf_list = self.get_non_system_ks_cf_list(loader_node, db_node_ip)
  File "/sct/sdcm/cluster.py", line 2952, in get_non_system_ks_cf_list
    .format(request_timeout, cmd, db_node_ip), verbose=False)
  File "/sct/sdcm/remote.py", line 916, in run
    watch_stdout_pattern=watch_stdout_pattern)
  File "/sct/sdcm/remote.py", line 857, in _run
    watch_stdout_pattern=watch_stdout_pattern)
  File "/sct/sdcm/remote.py", line 814, in ssh_run
    raise process.CmdError(cmd, sp.result)
CmdError: Command '/bin/ssh  -a -x   -o StrictHostKeyChecking=no -o UserKnownHostsFile=/var/tmp/tmpcKD17V -o BatchMode=yes -o ConnectTimeout=300 -o ServerAliveInterval=300 -l centos -p 22 -i /home/jenkins/.ssh/scylla-qa-ec2 10.0.88.72 "cqlsh --request-timeout=300 --no-color --execute \"SELECT keyspace_name, table_name from system_schema.tables\" 10.0.87.26"' failed (rc=1)

$ /bin/ssh  -a -x   -o StrictHostKeyChecking=no -o UserKnownHostsFile=/var/tmp/tmpcKD17V -o BatchMode=yes -o ConnectTimeout=300 -o ServerAliveInterval=300 -l centos -p 22 -i /home/jenkins/.ssh/scylla-qa-ec2 10.0.88.72 "cqlsh --request-timeout=300 --no-color --execute \"SELECT keyspace_name, table_name from system_schema.tables\" 10.0.87.26"
Connection error: ('Unable to connect to any servers', {'10.0.87.26': AuthenticationFailed('Remote end requires authentication.',)})

```

**Comment by bentsi** [comment]
@amoskong unfortunately, this PR is not backportable to 2019.1 (too many conflicts)

**Comment by amoskong** [comment]
On Sun, Jul 14, 2019 at 10:05 PM Bentsi <notifications@github.com> wrote:

> @amoskong <https://github.com/amoskong> unfortunately, this PR is not
> backportable to 2019.1 (too many conflicts)
>

Ok, we can have a patch for branch-2019.1 and add cql_auth for rest cqlsh
commands.
I will send one PR

**Comment by amoskong** [comment]
> Ok, we can have a patch for branch-2019.1 and add cql_auth for rest cqlsh commands. I will send one PR

PR for branch-2019.1:
- cqlsh: add cql_auth support for rest cqlsh commands #1157

**Comment by bentsi** [comment]
@amoskong this is part of new  `branch-2019-next` that will become 2019.1

**Pull request merged**